server/app/auth/CiviFormProfileData.java

```diff
@@ -70,6 +73,20 @@ public String getSessionId() {
     return getAttributeAsString(SESSION_ID);
   }
 
+  public void updateLastActivityTime(Clock clock) {
+    addAttribute(LAST_ACTIVITY_TIME, clock.instant().toEpochMilli());
+  }
+
+  /**
+   * Returns the timestamp of the last activity for this session. For backward compatibility with
+   * existing sessions created before session timeout feature, returns current time if the attribute
+   * is not present. This effectively gives legacy sessions a fresh activity timestamp when the
+   * timeout feature is first enabled.
+   */
+  public long getLastActivityTime(Clock clock) {
+    return (Long) getAttributes().getOrDefault(LAST_ACTIVITY_TIME, clock.instant().toEpochMilli());
+  }
```

Comment:
```suggestion
  public long getLastActivityTime() {
    return (Long) getAttributes().getOrDefault(LAST_ACTIVITY_TIME, clock.instant().toEpochMilli());
  }
```

(blocking) Please use the class level clock.